server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreCommitLevelLockManager.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.lockmanager;
+
+import java.io.IOException;
+
+/**
+ * An Interface that defines Commit Level Lock in Remote Store.
+ * @opensearch.internal
+ */
+public interface RemoteStoreCommitLevelLockManager {
```

Comment:
Agree, having these methods in `RemoteSegmentStoreDirectory` makes sense. I am trying to understand if we really need an interface here.

Reply:
+1 . I am also not sure about if we would have more implementations of this . Commit level locks looks like a class rather than interface .

Reply:
I think we can still have this interface because lets say we want to let master node, just use the lock manager related functionalities but not remote store using functionality, we can still pass RemoteSegmentStoreDirectory instance on LockManager Interface.